src/app/modals/complete-task-pop-up/complete-task-pop-up.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { TaskService } from 'src/app/shared-module/services/task.service';
+
+@Component({
+  selector: 'app-complete-task-pop-up',
+  templateUrl: './complete-task-pop-up.component.html',
+  styleUrls: ['./complete-task-pop-up.component.scss'],
+})
+export class CompleteTaskPopUpComponent {
+  completeTaskResult?: any;
+
+  constructor(private taskService: TaskService, private dialogRef: MatDialog) {}
+
+  onSubmit() {
+    // TODO
+    this.taskService.completeCurrentTask().subscribe({
+      next: () => {
```

Comment:
.pipe(catchError(val => { ___ ; return of(___)}))

Reply:
As I understood we need to use pipe in the serviece to filter Observable. And if we need pass error message to the component, we get it there from subscribe?

Reply:
we can still access it from the pipe, unless you have pipe(catchError) in taskService which already handled the error